rpc/jsonrpc/server/rpc_func.go

```diff
@@ -24,38 +24,54 @@ func RegisterRPCFuncs(mux *http.ServeMux, funcMap map[string]*RPCFunc, logger lo
 }
 
 // Function introspection
+const (
+	Cacheable = "cacheable"
+	Ws        = "ws"
+)
 
 // RPCFunc contains the introspected type information for a function
 type RPCFunc struct {
-	f        reflect.Value  // underlying rpc function
-	args     []reflect.Type // type of each function arg
-	returns  []reflect.Type // type of each return arg
-	argNames []string       // name of each argument
-	ws       bool           // websocket only
+	f        reflect.Value          // underlying rpc function
+	args     []reflect.Type         // type of each function arg
+	returns  []reflect.Type         // type of each return arg
+	argNames []string               // name of each argument
+	opts     map[string]interface{} // options of RPCFunc
 }
 
 // NewRPCFunc wraps a function for introspection.
 // f is the function, args are comma separated argument names
-func NewRPCFunc(f interface{}, args string) *RPCFunc {
-	return newRPCFunc(f, args, false)
+func NewRPCFunc(f interface{}, args string, options ...string) *RPCFunc {
```

Comment:
This is not exactly what I meant originally by using a functional options approach. The idea here would be to define a type:

```golang
type Option func(*RPCFunc)
```

that would allow you to define functional options as follows:

```golang
func Cacheable() Option {
    return func(r *RPCFunc) {
        r.cacheable = true
    }
}
```

assuming the `RPCFunc` struct looks like:

```golang
type RPCFunc struct {
    // ...
    cacheable bool
}
```

So the function signature here would be:

```golang
func NewRPCFunc(f interface{}, options ...Option) *RPCFunc {
    // ...
```

and when applying the options:

```golang
for _, opt := range options {
    opt(r)
}
```

Reply:
Thanks for pointing out the mistakes. Updated!